OrleansDashboard/Metrics/Grains/DashboardRemindersGrain.cs

```diff
@@ -9,24 +10,30 @@ namespace OrleansDashboard
 {
     public class DashboardRemindersGrain : Grain, IDashboardRemindersGrain
     {
+        private static readonly ReminderResponse EmptyReminders = new ReminderResponse
+        {
+            Reminders = new ReminderInfo[0]
+        };
+
         private readonly IReminderTable _reminderTable;
 
-        public DashboardRemindersGrain(IReminderTable reminderTable)
+        public DashboardRemindersGrain(IServiceProvider serviceProvider)
         {
-            _reminderTable = reminderTable;
+            _reminderTable = serviceProvider.GetService(typeof(IReminderTable)) as IReminderTable;
         }
 
         public async Task<Immutable<ReminderResponse>> GetReminders(int pageNumber, int pageSize)
         {
+            if (_reminderTable == null)
+            {
+                return EmptyReminders.AsImmutable();
```

Comment:
Just an idea: Instead `EmptyReminders` could be this `Immutable<ReminderResponse>` directly and thus prevent calls to `AsImmutable` when needed. Additionally it would emphasize that `EmptyReminders` itself should stay unchanged throughout the app's lifetime.

Reply:
Yeah, I was thinking about it and then realized that immutable is a struct, so there is no allocation. But I will change it.